formula-test/src/main/java/com/instacart/formula/test/TestFormula.kt

```diff
@@ -62,9 +71,8 @@ abstract class TestFormula<Input, Output> :
             output = state.output,
             actions = context.actions {
                 Action.onTerminate().onEvent {
-                    transition {
-                        stateMap.remove(state.key)
-                    }
+                    stateMap.remove(state.uniqueIdentifier)
```

Comment:
Moving the side-effect here so it happens immediately and not after all the state changes are processed.